tests/integration/local/invoke/test_integration_cli_images.py

```diff
@@ -399,3 +399,88 @@ def test_invoke_with_error_during_image_build(self):
             "Error: Error building docker image: pull access denied for non-existing-image",
         )
         self.assertEqual(process.returncode, 1)
+
+
+class TestDeleteOldRapidImages(InvokeIntegBase):
+    template = Path("template_image.yaml")
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestDeleteOldRapidImages, cls).setUpClass()
+        cls.client = docker.from_env()
+        cls.repo = "sam-test-lambdaimage"
+        cls.tag = f"{cls.repo}:v1"
+        cls.test_data_invoke_path = str(Path(__file__).resolve().parents[2].joinpath("testdata", "invoke"))
+        # Directly build an image that will be used across all local invokes in this class.
+        for log in cls.client.api.build(
+            path=cls.test_data_invoke_path, dockerfile="Dockerfile", tag=cls.tag, decode=True, nocache=True
+        ):
+            print(log)
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.client.api.remove_image(cls.tag)
+        except APIError:
+            pass
+
+    def setUp(self):
+        self.old_rapid_image_tags = [f"{self.repo}:rapid-0.00.01", f"{self.repo}:rapid-0.00.02"]
+        for tag in self.old_rapid_image_tags:
+            for log in self.client.api.build(
+                path=self.test_data_invoke_path, dockerfile="Dockerfile", tag=tag, decode=True, nocache=True
+            ):
+                print(log)
+        self.new_rapid_image_tag = f"{self.repo}:rapid-{version}"
+
+    def tearDown(self):
+        for tag in self.old_rapid_image_tags + [self.new_rapid_image_tag]:
+            try:
+                self.client.api.remove_image(tag)
+            except APIError:
+                pass
+
+    @pytest.mark.flaky(reruns=3)
+    def test_building_new_rapid_image_removes_old_rapid_images(self):
```

Comment:
Suggest to test a case that it doesn't remove the `:rapid-` tagged images from the other repos.

Reply:
Thanks for the suggestion, I'll add an integration test for this